src/Validation/Validator.php

```diff
@@ -500,20 +514,62 @@ public function requirePresence($field, $mode = true, $message = null)
      * Because this and `notEmpty()` modify the same internal state, the last
      * method called will take precedence.
      *
-     * @param string $field the name of the field
+     * @param string|array $field the name of the field or list of fields
      * @param bool|string|callable $when Indicates when the field is allowed to be empty
      * Valid values are true (always), 'create', 'update'. If a callable is passed then
      * the field will allowed to be empty only when the callback returns true.
      * @return $this
      */
     public function allowEmpty($field, $when = true)
     {
-        $this->field($field)->isEmptyAllowed($when);
+        if (!is_array($field)) {
+            $field = $this->_convertValidatorToArray($field, ['when'], [$when]);
+        }
+
+        foreach ($field as $fieldName => $setting) {
+            $settings = $this->_convertValidatorToArray($fieldName, ['when'], [$when], $setting);
+            $fieldName = current(array_keys($settings));
+
+            $this->field($fieldName)->isEmptyAllowed($settings[$fieldName]['when']);
+        }
+
         return $this;
     }
 
     /**
-     * Sets a field to require a non-empty value.
+     * Converts validator to fieldName => $setting array
+     *
+     * @param string $fieldName name of field
+     * @param array $settingKeys keys that will be used to create default settings
+     * @param array $settingsValues values that will be used to create default settings
+     * @param array $setting settings from data
+     * @return array
+     */
+    protected function _convertValidatorToArray($fieldName, $settingKeys = [], $settingsValues = [], $setting = [])
+    {
+        if (is_string($setting)) {
```

Comment:
string is not mentioned in doc block. Also, variable might better be $settings maybe?